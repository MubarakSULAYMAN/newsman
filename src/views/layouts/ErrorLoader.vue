<template>
  <v-container class="error-loader rounded-lg elevation-24" v-if="isErrorLoading">
    <v-row class="fill-height" align-content="center" justify="center">
      <v-col class="subtitle-1 text-center" cols="12">
        Getting your request done
      </v-col>
      <v-col cols="6">
        <v-progress-linear
          color="indigo accent-4"
          indeterminate
          rounded
          height="6"
        ></v-progress-linear>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'ErrorLoader',

  computed: {
    ...mapGetters(['isErrorLoading']),
  },
};
</script>

<style scoped>
.error-loader {
  position: fixed;
  top: 50%;
  left: 50%;
  background-color: white;
  transform: translate(-50%, -50%);
  /* z-index: 1111; */
  /* background-image: linear-gradient(rgba(75, 0, 130, 0.5), rgba(75, 0, 130, 0.5)); */
}
</style>
