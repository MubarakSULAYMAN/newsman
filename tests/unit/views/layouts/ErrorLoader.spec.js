import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import ErrorLoader from '@/views/layouts/ErrorLoader.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Error Noification', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  test('It displays "" for errorInfo and "Error message expected Here." for errorMessage.', () => {
    const $store = {
      getters: {
        isErrorLoading: () => false,
      },
    };

    const wrapper = mount(ErrorLoader, {
      localVue,
      vuetify,
      mocks: {
        $store,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
