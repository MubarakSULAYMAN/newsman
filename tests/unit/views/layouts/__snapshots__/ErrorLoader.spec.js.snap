// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Noification It displays "" for errorInfo and "Error message expected Here." for errorMessage. 1`] = `
<div class="container error-loader rounded-lg elevation-24">
  <div class="row fill-height justify-center align-content-center">
    <div class="subtitle-1 text-center col col-12">
      Getting your request done
    </div>
    <div class="col col-6">
      <div role="progressbar" aria-valuemin="0" aria-valuemax="100" class="v-progress-linear v-progress-linear--rounded theme--light" style="height: 6px;">
        <div class="v-progress-linear__background indigo accent-4" style="opacity: 0.3; left: 0%; width: 100%;"></div>
        <div class="v-progress-linear__buffer"></div>
        <transition-stub name="fade-transition">
          <div class="v-progress-linear__indeterminate v-progress-linear__indeterminate--active">
            <div class="v-progress-linear__indeterminate long indigo accent-4"></div>
            <div class="v-progress-linear__indeterminate short indigo accent-4"></div>
          </div>
        </transition-stub>
      </div>
    </div>
  </div>
</div>
`;
